src/js/shot-generator/IK/objects/IkObjects/IkObject.js

```diff
@@ -0,0 +1,243 @@
+const {IK, IKJoint}  = require("../../core/three-ik");
+const THREE = require( "three");
+const {setZDirecion, setReverseZ} = require( "../../utils/axisUtils");
+const ChainObject = require( "./ChainObject");
+const SkeletonUtils = require("../../utils/SkeletonUtils");
+const IKSwitcher = require("./IkSwitcher");
+require("../../utils/Object3dExtension");
+
+class IkObject
+{
+    constructor()
+    {
+        if(new.target === IkObject)
+        {
+            throw new TypeError("Cannot construct abstract IkObject directly");
+        }
+        this.bonesDelta = {};
+        this.ikSwitcher = null;
+        this.isEnabledIk = false;
+        this.clonedObject = null;
+        this.controlTargets = [];
+        this.originalObject = null;
+        this.applyingOffset = false;
+        this.isRotation = false;
+        this.scene = null;  
+    }
+
+    //#region External Methods
+    // Takes skeleton and target for it's limbs
+    initObject(scene, objectSkeleton, controlTargets)
+    {
+        this.ik = new IK();
+        this.scene = scene;
+        let chains = [];
+        let clonedSkeleton = SkeletonUtils.clone(objectSkeleton);
+        this.clonedObject = clonedSkeleton;
+        this.originalObject = objectSkeleton;
+
+        this.ikSwitcher = new IKSwitcher(objectSkeleton, clonedSkeleton);
+
+        this.rigMesh = clonedSkeleton.children[1];
+        let rigMesh = this.rigMesh;
+
+        this.controlTargets = controlTargets;
+        this.addParentToControl(objectSkeleton.uuid);
+        let chainObjects = [];
+        this.chainObjects = chainObjects;
+        this.hipsControlTarget = this.controlTargets[5];
+
+        chainObjects.push(new ChainObject("Spine", "Head", this.controlTargets[0]));
+        chainObjects.push(new ChainObject("LeftArm", "LeftHand", this.controlTargets[1]));
+        chainObjects.push(new ChainObject("RightArm", "RightHand", this.controlTargets[2]));
+        chainObjects.push(new ChainObject("LeftUpLeg", "LeftFoot", this.controlTargets[3]));
+        chainObjects.push(new ChainObject("RightUpLeg", "RightFoot", this.controlTargets[4]));
+
+        //Fixing female-adult spine deformation
```

Comment:
Could we fix this in the model file instead?

Reply:
Yeah, that will be great. Every other character model works fine except for a female one.

Reply:
What would need to change in the model file? Bone #2 / Spine1 needs to have a rotation of [0, 0, 0]?